diff --git a/kcm/qml/OutputView.qml b/kcm/qml/OutputView.qml
index 0051592..9d1b342 100644
--- a/kcm/qml/OutputView.qml
+++ b/kcm/qml/OutputView.qml
@@ -318,12 +318,10 @@ Flickable {
 
             if (output.output.enabled) {
                 enabledCount++;
-                if (enabledCount > 0) {
-                    break;
-                }
             }
         }
 
+        tooManyActiveOutputs.opacity = ((enabledCount > virtualScreen.maxActiveOutputsCount) ? 1.0 : 0.0);
         noActiveOutputsWarning.opacity = (enabledCount > 0) ? 0.0 : 1.0;
     }
 
diff --git a/kcm/qml/main.qml b/kcm/qml/main.qml
index 3384d08..0bea88d 100644
--- a/kcm/qml/main.qml
+++ b/kcm/qml/main.qml
@@ -124,6 +124,15 @@ Item {
                 icon: "dialog-warning";
                 text: i18n("Warning: There are no active outputs!");
             }
+
+            Tip {
+
+                id: tooManyActiveOutputs;
+                objectName: "tooManyActiveOutputs";
+
+                icon: "dialog-error";
+                text: i18n("Your system only supports up to %1 active screens", virtualScreen ? virtualScreen.maxActiveOutputsCount : 1);
+            }
         }
 
         IconButton {
diff --git a/kcm/src/kcm_kscreen.cpp b/kcm/src/kcm_kscreen.cpp
index ea1a490..ed9560a 100644
--- a/kcm/src/kcm_kscreen.cpp
+++ b/kcm/src/kcm_kscreen.cpp
@@ -167,7 +167,7 @@ void KCMKScreen::load()
     }
     QMetaObject::invokeMethod(outputView, "reorderOutputs");
 
-    connect(outputView, SIGNAL(outputChanged()), SLOT(changed()));
+    connect(outputView, SIGNAL(outputChanged()), SLOT(outputChanged()));
     connect(outputView, SIGNAL(moveMouse(int,int)), SLOT(moveMouse(int,int)));
     connect(outputView, SIGNAL(outputMousePressed()), SLOT(outputMousePressed()));
     connect(outputView, SIGNAL(outputMouseReleased()), SLOT(outputMouseReleased()));
@@ -209,6 +209,14 @@ void KCMKScreen::save()
         }
     }
 
+    if (!Config::canBeApplied(m_config)) {
+        KMessageBox::information(this,
+            i18n("Sorry, your configuration could not be applied.\n\n"
+                 "Common reasons are that the overall screen size is too big, or you enabled more displays than supported by your GPU."),
+                 i18n("Unsupported configuration"));
+        return;
+    }
+
     /* Store the current config, apply settings */
     m_config->setConfig(m_config);
 }
@@ -284,4 +292,17 @@ void KCMKScreen::outputMouseReleased()
     m_declarativeView->setCursor(Qt::ArrowCursor);
 }
 
+void KCMKScreen::outputChanged()
+{
+    if (!Config::canBeApplied(m_config)) {
+        setButtons(KCModule::Default);
+        changed(false);
+    } else {
+        setButtons(KCModule::Default | KCModule::Apply);
+        changed(true);
+    }
+
+}
+
+
 #include "kcm_kscreen.moc"
diff --git a/kcm/src/kcm_kscreen.h b/kcm/src/kcm_kscreen.h
index 97e770c..16d9c76 100644
--- a/kcm/src/kcm_kscreen.h
+++ b/kcm/src/kcm_kscreen.h
@@ -50,6 +50,7 @@ private Q_SLOTS:
 
     void outputMousePressed();
     void outputMouseReleased();
+    void outputChanged();
 
 private:
     KScreen::Config *m_config;
